components/brave_shields/browser/site_specific_script_service.h

```diff
@@ -0,0 +1,76 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_SHIELDS_BROWSER_SITE_SPECIFIC_SCRIPT_SERVICE_H_
```

Comment:
It doesn't seem like `brave_shields` is the right place for this. Maybe `brave/browser/` or a new component like maybe `components/site_scripts` @bbondy ?

Reply:
in particular since this is going to be used by rewards it will not be connected to brave_shield settings